src/include/compat.h

```diff
@@ -307,4 +307,12 @@ int lchown(const char *path, uid_t owner, gid_t group);
 
 #endif /* WIN32 */
 
+#ifdef __MINGW32__
```

Comment:
We were considering adding MSVC support later on, which would also allow us to use clang with the msvc driver. We were mostly concerned about long standing mingw bugs but we managed to work around them so it's not that urgent anymore.

I can avoid this kind of checks for now, if you prefer.

Reply:
i want to minimize the use of `#ifdef` unless we really need to. otherwise, in the long run, our code will be full of these kind of checks even if some of them are not used.

please void them unless they will be maintained / tested.

Reply:
Sure, makes sense.